### ISSUE TITLE
Medspacy Installation Issue v1.1.2

### ISSUE THREAD

**Issue by pyturn** [medspacy/medspacy#261]
We used to have medspacy `v1.1.2` in our application that used to run fine without any issues till last week. Now we are not able to install the  medspacy `v1.1.2` version and its showing unexplainable behaviour in different environments - 

**In Docker Image (Main Issue)**  - We are installing medspacy `v1.1.2` which used to run till last week. Today it started giving the error as shown in below screenshot. 

![image](https://github.com/medspacy/medspacy/assets/25935364/86259917-3f09-444c-bbd8-1c8a8fb56405)

**Debugging Steps**

- Virtual environment created on (Local Machine) : Medspacy `v1.1.2` installed successfully without any issues. 

- On Google colab :  Its showing the error as shown in below screenshot. 

<img width="655" alt="image" src="https://github.com/medspacy/medspacy/assets/25935364/85e2cc63-b7a8-4c51-b1ac-6097ef87dbaa">

- We cross checked the tag `v1.1.2` across the list of tags available here - https://github.com/medspacy/medspacy/tags (And surprisingly we did not find `v1.1.2` which was present earlier. It also available on https://pypi.org/search/?q=medspacy and released on May 8 , 2023) 

We are facing the installation issue in Docker image. And it also shows installation related issue on Google colab. (However it used to run fine till last week).

**Comment by jianlins** [comment]
Thanks for reporting this issue. I set a new pull request to solve it.
https://github.com/medspacy/medspacy/pull/262
Once it's approved, we will release a new minor version updates

Jianlin

**Comment by gp48maz1** [comment]
@kgweber-um I tried the change to the `pyproject.toml` file but no luck same error -- any other recommendations?

**Comment by editnori** [comment]
> @kgweber-um I tried the change to the `pyproject.toml` file but no luck same error -- any other recommendations?

That setuptools change is correct, but it didnt work for me either until i realized it had to do with the PyrRUSH dependency. I had python 3.12 which wasnt compatible i installed python 3.8.10 and reran it and it worked. hope that helps!

**Comment by kgweber-um** [comment]
Thanks for that insight, @editnori  - I'm running 3.11 and didn't run into that particular blocker.

**Comment by mpasternak** [comment]
I'm on Python 3.11 and I'm unable to install medspacy with pip.

**Comment by mpasternak** [comment]
You need to add 

```
readme = "README.md"
```
to pyproject.toml, section `[project]`

Arę maintainers of this project alive? I hope. If so, send me a quick note and I will gladly prepare a PR!

**Comment by kgweber-um** [comment]
This tool's a workhorse for me: I fervently hope someone still loves it.

**Comment by editnori** [comment]
@kgweber-um What are some things you have done with medspacy thus far? Would love to hear more about your projects!

**Comment by kgweber-um** [comment]
Oh @editnori if you get me going, I'll never stop talking. Let's not derail this very important thread, but I'm happy to discuss in another venue.

**Comment by editnori** [comment]
> Oh @editnori if you get me going, I'll never stop talking. Let's not derail this very important thread, but I'm happy to discuss in another venue.

Totally, here is my email feel free to message me there: layth.qassem@vanderbilt.edu

**Comment by allanckw** [comment]
Hi, is this resolved? I am still getting the error with Python 3.9.13 and i cannot install it

**Comment by burgersmoke** [comment]
We just prepared medspacy `1.1.5` which is now published to PyPI.  

Could someone experiencing these installation issues try 1.1.5 and let us know if you are unblocked now.

Thank you all for your patience!

**Comment by editnori** [comment]
I can take a look at it today after work!

**Comment by jianlins** [comment]
I will close it for now. If it is still an issue, we can reopen it.

**Comment by kgweber-um** [comment]
(I know the issue's closed - will let you keep it that way if you want to)

I just installed medspacy 1.1.5 via `pip` and the install was almost clean, but I did get more `setuptools` errors as `unqlite` was installing. It rolled to a backup approach and I have medspacy on board, but the stream of errors might be unsettling to someone new to the project. Is there a dependency that needs to be updated?

```
Building wheels for collected packages: medspacy, unqlite
  Building wheel for medspacy (pyproject.toml) ... done
  Created wheel for medspacy: filename=medspacy-1.1.5-py3-none-any.whl size=147959 sha256=d419f4d48a3acc6c3ff59a4df59294cbf4342a35dc13ff6e5a68d4c91cd931dd
  Stored in directory: /Users/[kgweber-um]/Library/Caches/pip/wheels/d3/08/88/26282a4ec76d786bd9289dfee36947149967891c6cc2fda687
  Building wheel for unqlite (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [94 lines of output]
      /private/var/folders/v3/xvdkctpx75z05zblz25nw7l80000gn/T/pip-install-799bum3g/unqlite_54eca13d477344cf89206d985c2c6115/setup.py:11: UserWarning: Cython not installed, using pre-generated C source file.
        warnings.warn('Cython not installed, using pre-generated C source file.')
      /Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      running bdist_wheel
      running build
      running build_ext
      building 'unqlite' extension
      creating build
      creating build/temp.macosx-10.9-x86_64-cpython-310
      creating build/temp.macosx-10.9-x86_64-cpython-310/src
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/include -fPIC -O2 -isystem /Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/include -DUNQLITE_ENABLE_THREADS=1 -I/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/include/python3.10 -c src/unqlite.c -o build/temp.macosx-10.9-x86_64-cpython-310/src/unqlite.o
      src/unqlite.c:4886:6: warning: variable 'rc' set but not used [-Wunused-but-set-variable]
              int rc;
                  ^
      src/unqlite.c:8555:6: warning: variable 'rc' set but not used [-Wunused-but-set-variable]
              int rc;
                  ^
      src/unqlite.c:12254:8: warning: variable 'zExtra' set but not used [-Wunused-but-set-variable]
              char *zExtra;
                    ^
      src/unqlite.c:18413:8: warning: variable 'nLine' set but not used [-Wunused-but-set-variable]
              sxu32 nLine;
                    ^
      src/unqlite.c:19876:11: warning: variable 'pCur' set but not used [-Wunused-but-set-variable]
              SyToken *pCur, *pIn;  /* Token stream */
                       ^
      src/unqlite.c:29176:9: warning: variable 'zExtra' set but not used [-Wunused-but-set-variable]
        char *zExtra;
              ^
      src/unqlite.c:29928:9: warning: variable 'rc' set but not used [-Wunused-but-set-variable]
              sxi32 rc;
                    ^
      src/unqlite.c:36149:67: warning: comparison of integers of different signs: 'jx9_int64' (aka 'long long') and 'unsigned long' [-Wsign-compare]
                      n = pStream->xRead(pDev->pHandle, zBuf, (nMaxLen > 0 && nMaxLen < sizeof(zBuf)) ? nMaxLen : sizeof(zBuf));
                                                                              ~~~~~~~ ^ ~~~~~~~~~~~~
      src/unqlite.c:37025:29: warning: comparison of integers of different signs: 'jx9_int64' (aka 'long long') and 'unsigned long' [-Wsign-compare]
                              (nMaxlen > 0 && (nMaxlen < sizeof(zBuf))) ? nMaxlen : sizeof(zBuf));
                                               ~~~~~~~ ^ ~~~~~~~~~~~~
      src/unqlite.c:27310:19: warning: unused function 'SyMemBackendDisbaleMutexing' [-Wunused-function]
      JX9_PRIVATE sxi32 SyMemBackendDisbaleMutexing(SyMemBackend *pBackend)
                        ^
      src/unqlite.c:38961:22: warning: unused variable 'null_vfs' [-Wunused-const-variable]
      static const jx9_vfs null_vfs = {
                           ^
      11 warnings generated.
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/include -fPIC -O2 -isystem /Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/include -DUNQLITE_ENABLE_THREADS=1 -I/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/include/python3.10 -c unqlite.c -o build/temp.macosx-10.9-x86_64-cpython-310/unqlite.o
      creating build/lib.macosx-10.9-x86_64-cpython-310
      clang -bundle -undefined dynamic_lookup -Wl,-rpath,/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib -L/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib -Wl,-rpath,/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib -L/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib build/temp.macosx-10.9-x86_64-cpython-310/src/unqlite.o build/temp.macosx-10.9-x86_64-cpython-310/unqlite.o -o build/lib.macosx-10.9-x86_64-cpython-310/unqlite.cpython-310-darwin.so
      /Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      installing to build/bdist.macosx-10.9-x86_64/wheel
      running install
      running install_lib
      creating build/bdist.macosx-10.9-x86_64
      creating build/bdist.macosx-10.9-x86_64/wheel
      copying build/lib.macosx-10.9-x86_64-cpython-310/unqlite.cpython-310-darwin.so -> build/bdist.macosx-10.9-x86_64/wheel
      running install_egg_info
      running egg_info
      writing unqlite.egg-info/PKG-INFO
      writing dependency_links to unqlite.egg-info/dependency_links.txt
      writing requirements to unqlite.egg-info/requires.txt
      writing top-level names to unqlite.egg-info/top_level.txt
      reading manifest file 'unqlite.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.o' found anywhere in distribution
      warning: no previously-included files matching '*.so.0.1' found anywhere in distribution
      adding license file 'LICENSE'
      writing manifest file 'unqlite.egg-info/SOURCES.txt'
      Copying unqlite.egg-info to build/bdist.macosx-10.9-x86_64/wheel/unqlite-0.9.6-py3.10.egg-info
      running install_scripts
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/v3/xvdkctpx75z05zblz25nw7l80000gn/T/pip-install-799bum3g/unqlite_54eca13d477344cf89206d985c2c6115/setup.py", line 34, in <module>
          setup(
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 968, in run_commands
          self.run_command(cmd)
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 328, in run
          impl_tag, abi_tag, plat_tag = self.get_tag()
        File "/Users/[kgweber-um]/.conda/envs/RiskyAlcoholUse/lib/python3.10/site-packages/wheel/bdist_wheel.py", line 278, in get_tag
          assert tag in supported_tags, "would build wheel with unsupported tag {}".format(tag)
      AssertionError: would build wheel with unsupported tag ('cp310', 'cp310', 'macosx_10_9_x86_64')
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for unqlite
  Running setup.py clean for unqlite
Successfully built medspacy
Failed to build unqlite
Installing collected packages: pysimstring, cymem, wasabi, unidecode, typer, spacy-loggers, spacy-legacy, smart-open, pysbd, pydantic, murmurhash, langcodes, Cython, catalogue, blis, unqlite, srsly, quicksectx, preshed, pathy, PyFastNER, confection, thinc, spacy, PyRuSH, medspacy-quickumls, medspacy
  Running setup.py install for unqlite ... done
  DEPRECATION: unqlite was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. A possible replacement is to fix the wheel build issue reported above. Discussion can be found at https://github.com/pypa/pip/issues/8368
```

**Comment by burgersmoke** [comment]
@kgweber-um  I think the issue you report above is likely a separate issue.  I think that your issue is probably most related to this line right here:

`AssertionError: would build wheel with unsupported tag ('cp310', 'cp310', 'macosx_10_9_x86_64')`

It looks like you are using Python 3.10?  

Also, It looks like this is Mac OSX 10.9.  Is that right?

By any chance have you tried using an older version of Python like 3.9 or 3.8?  I ask since OSX 10.9 is a bit old now and it looks like others online have had issues with Python versions 3.10 or higher.

Could you add a new ticket with the information above as well as your OS and Python version details?
